src/ChannelServer/World/Inventory.cs

```diff
@@ -232,12 +271,30 @@ public int AddSilent(Item item)
 		/// <return>Index of the item.</return>
 		public int Add(Item item, InventoryAddType addType)
 		{
-			var index = this.AddSilent(item);
+			Dictionary<int, int> indexes = this.AddSilent(item);
+
+			var lastItem = indexes.Last();
+			int lastIndex = lastItem.Key;
+
+			foreach (var itemObject in indexes)
+			{
+				int index = itemObject.Key;
+				int diff = itemObject.Value;
+				
+				if (index == lastIndex)
+				{
+					addType = InventoryAddType.PickUp;
```

Comment:
If i add not stackable item it should be new always - yes.
If add some pieces to stack it shouldn't be new - no.
But last stack can be new sometime and always new for nonstackable.

Not sure how it should be.